rancher/distros-test-framework
# refactor: improve validate functions to be dynamic

#### Proposed Changes ####



- improve validate function to enable consumer decide which value to use on timeout and tickers




#### Types of Changes ####

- Refactor on validate function




#### Testing ####


Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.
- n/a

2. Verify code lint; we should not have errors.
- ok 

3. Update the documentation if needed.

  -n/a
4. Update makefile and docker run if adding new tests.

  -n/a
5. Run your tests at least 4 times with all configurations needed and possible.

- ok
6. If needed test with different os types.

- n/a 
#### Linked Issues ####

- https://github.com/rancher/qa-tasks/issues/964 


#### Further Comments ####
